### DISCUSSION TITLE
Time and space dependent material properties

### DISCUSSION THREAD

**Discussion by petr-henys** [OpenRadioss/OpenRadioss#863] [Q&A]
Hi,
i have Young modulus as a function of coordinates and time. Is there a way how to handle such function in OpenRadioss?

Thanks,

Petr

**Comment by MarianBulla** [comment]
Hi, 
There is no out-of-the-box functionality for this, in OpenRadioss. 
But...you can create a new material model, where this functionality is implemented. 
For the second option (time dependant Young's modulus) it is quite simple: You just need to provide a function (time vs. Young's modulus) into your material model and compute the Young's modulus dependant on the actual time.
The first option is more difficult, because of which coordinates ?
Your part consists usually out of many nodes. Which node should be used ?

Hope it helps you ?

Thanks and best regards,
Marian

**Comment by petr-henys** [comment]
Understand,
the Young's modulus is a funtion of coordinates: E = f(x,y,z) --see figure:
![image](https://user-images.githubusercontent.com/22882861/205925577-8b555341-2a6b-4d46-9ef4-a3bca723518c.png)
The function can be evaluated at nodes/integration points or element barycentres - depending on how the FE software handles it.

**Comment by MaciekWronski** [comment]
Hi Petr,

Ok, what I figure out from your picture is that you rather need an  initialization of Young modulus depending on the position in your bone. These values could be mapped to the integration points rather than nodes, for simplicity reason. Then Young modulus would be treated as an internal history variable instead of constant material parameter. This way your material model does not need any heavy modifications.

Anyway, there’s no automatic tool in radioss to perform this kind of mapping but a development of it should be obviously possible.
What will be left to material model itself would be only a time dependency, which is very easy to implement. Of course, assuming that this time scale factor is  independent on the spatial position itself.
Do I understand correctly what you intend to do ?

Kind regards,
Maciek


Maciek Wronski
Director, Software Development

Altair | Nasdaq : ALTR
***@***.******@***.***> | altair.com<https://nam02.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.altair.com%2F&data=04%7C01%7Cmaciek%40altair.com%7C4d884d8253134ee2501708d98a447ff8%7C2bae5b570eb848fbba47990259da89d2%7C0%7C0%7C637692849078713773%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Y5BqP60FcZBmWsHgnzLuvUyrI1XIYzWeeYVViIYEN9U%3D&reserved=0>
Connect with us: LinkedIn<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.linkedin.com%2Fcompany%2Faltair-engineering%2F&data=04%7C01%7Cmaciek%40altair.com%7C4d884d8253134ee2501708d98a447ff8%7C2bae5b570eb848fbba47990259da89d2%7C0%7C0%7C637692849078723728%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ikf9fUmAzEGgOiqK0xRZoCtgHCTM%2BUkh%2FuGXJdCJSNs%3D&reserved=0> | Twitter<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Ftwitter.com%2Faltair_inc&data=04%7C01%7Cmaciek%40altair.com%7C4d884d8253134ee2501708d98a447ff8%7C2bae5b570eb848fbba47990259da89d2%7C0%7C0%7C637692849078733692%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=CXoGju4aANw%2Fv1N3xVjM%2BRQVbVZ8fxs8RQ8WQnqUTv0%3D&reserved=0> | Facebook<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.facebook.com%2Faltairengineering%2F&data=04%7C01%7Cmaciek%40altair.com%7C4d884d8253134ee2501708d98a447ff8%7C2bae5b570eb848fbba47990259da89d2%7C0%7C0%7C637692849078733692%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=cVHEQJGlGo2OnWobs7c2beYVYdHvyRcRxoUowMg8Xuo%3D&reserved=0>
#OnlyForward

De : Petr Henyš ***@***.***>
Envoyé : mardi 6 décembre 2022 14:33
À : OpenRadioss/OpenRadioss ***@***.***>
Cc : Subscribed ***@***.***>
Objet : [EXTERNAL] Re: [OpenRadioss/OpenRadioss] Time and space dependent material properties (Discussion #863)


CAUTION: This email originated from outside the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.

Understand,
the Young's modulus is a funtion of coordinates: E = f(x,y,z) --see figure:
[image]<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F22882861%2F205925577-8b555341-2a6b-4d46-9ef4-a3bca723518c.png&data=05%7C01%7Cmaciek%40altair.com%7C0387cb9f41cf4f6ae44108dad78e5add%7C2bae5b570eb848fbba47990259da89d2%7C0%7C0%7C638059303650821205%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Ww0uSJ5qcbWGJDCbrwMcXWj433fCTi1uxbZYmG0ryTE%3D&reserved=0>
The function can be evaluated at nodes/integration points or element barycentres - depending on how the FE software handles it.

**Comment by MarianBulla** [comment]
Hi,
yes, the mapping is possible to be done with pro-processors like HyperMesh or HyperCrash or scritps.
Thanks and best regards,
Marian

**Comment by petr-henys** [comment]
Hi both,
sounds very promising. The time dependence does not interact with spatial coords. Beside spatial varying elasticity, the bone structure is highly anisotropic and this anisotropy is defined by spatially varying tensor-valued function (local coordinates field). Similarly, parameters for plasticity models or damage models are defined as space function as well. 

Should i implement these functions on the scripting level or do I need to dive into C++/fortran? Infact i am not really expert in these languages...

**Reply by MarianBulla** [reply]
Hi Petr, 
for such local material properties (local orthotropy and stiffness) we developed the material LAW93 (/MAT/CONVERSE/).
Maybe it is already sufficient for you ?
If the mesh is fine enough, I would propose to use an isotropic material law (like /MAT/LAW36/), where the individual properties are just scalled and individual damage values are applied.

![image](https://user-images.githubusercontent.com/105412502/206128650-d2ff2ec5-419f-476d-8142-e3a18bc4cf14.png)

**Comment by petr-henys** [comment]
Hi both,
What is the difference between these two laws? I still need clarification about how Openradioss handles spatial functions and how to implement them correctly...
What do you mean by using an isotropic material model if the mesh is fine enough? Should I transform the material tensor myself and then use it in OpenRadioss?

thanks,
Petr
